src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/DefaultLSPDiagnosticsProvider.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using Microsoft.VisualStudio.LanguageServer.ContainedLanguage;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor.HtmlCSharp
+{
+    [Shared]
+    [Export(typeof(LSPDiagnosticsProvider))]
+    internal class DefaultLSPDiagnosticsProvider : LSPDiagnosticsProvider
+    {
+        private readonly LSPRequestInvoker _requestInvoker;
+
+        [ImportingConstructor]
+        public DefaultLSPDiagnosticsProvider(LSPRequestInvoker requestInvoker)
+        {
+            if (requestInvoker is null)
+            {
+                throw new ArgumentNullException(nameof(requestInvoker));
+            }
+
+            _requestInvoker = requestInvoker;
+        }
+
+        public override async Task<RazorDiagnosticsResponse> ProcessDiagnosticsAsync(
+            RazorLanguageKind languageKind,
+            Uri razorDocumentUri,
+            Diagnostic[] diagnostics,
+            LanguageServerMappingBehavior mappingBehavior,
```

Comment:
Will we ever not use "inclusive" for diagnostics?

Reply:
Don't think so, but wanted to keep the option open. Given we're only using inclusive, going to remove it from the request params for now.